tests/general/AnnotationBearTest.py

```diff
@@ -22,47 +22,159 @@ def setUp(self):
 
     def test_single_line_string(self):
         text = ["'from start till the end with #comments'\n", ]
-        compare = (SourceRange.from_absolute_position(
-                                    'F',
-                                    AbsolutePosition(text, 0),
-                                    AbsolutePosition(text, len(text[0]) - 2)),)
+        compare_full = SourceRange.from_absolute_position(
+            'F',
+            AbsolutePosition(text, 0),
+            AbsolutePosition(text, len(text[0]) - 2))
+
+        compare_content = SourceRange.from_absolute_position(
+            'F',
+            AbsolutePosition(text, 1),
+            AbsolutePosition(text, len(text[0]) - 3))
+
+        compare_start = SourceRange.from_absolute_position(
+            'F',
+            AbsolutePosition(text, 0),
+            AbsolutePosition(text, 0))
+
+        compare_end = SourceRange.from_absolute_position(
```

Comment:
@rhemon Better data and algorithm abstractions are a general issue with this PR :/